crates/iceberg-catalog/src/catalog/tables.rs

```diff
@@ -819,6 +842,19 @@ impl<C: Catalog, A: AuthZHandler, S: SecretStore>
             ));
         }
 
+        for ((update, (_, table_uuid)), response) in updates
+            .into_iter()
+            .zip(&event_table_ids)
+            .zip(&commit_response)
+        {
+            state
+                .v1_state
+                .contract_verifiers
+                .check(&update, *table_uuid, &response.previous_table_metadata)
+                .await?
```

Comment:
There might be quite a few updates here - I would prefer concurrent execution and try_join

Reply:
It doesn't quite fit here, but I am also a bit hesitant about awaiting the emit_change_event block.
We currently implement a "send and forget" approach. But right now, if the event queue is offline, we are blocking the client response until the event queue gets a timeout.
What's your feeling about emitting events non-blocking?

Reply:
Do we expect this endpoint to be called a lot? How many changes do we expect per call? 

We could separate it out via a channel and have some background task shoveling it into nats.

Reply:
> There might be quite a few updates here - I would prefer concurrent execution and try_join

made it futures + `try_join_all`

Reply:
> Do we expect this endpoint to be called a lot? How many changes do we expect per call?
> 
> We could separate it out via a channel and have some background task shoveling it into nats.

https://github.com/hansetag/iceberg-catalog/issues/86